xarray/backends/netCDF4_.py

```diff
@@ -343,11 +343,22 @@ def set_dimension(self, name, length, is_unlimited=False):
             dim_length = length if not is_unlimited else None
             self.ds.createDimension(name, size=dim_length)
 
+    def is_list_of_strings(self, value):
+        if (np.asarray(value).dtype.kind in ['U', 'S'] and
+                np.asarray(value).size > 1):
+            return True
+        else:
+            return False
+
     def set_attribute(self, key, value):
         with self.ensure_open(autoclose=False):
             if self.format != 'NETCDF4':
                 value = encode_nc3_attr_value(value)
-            self.ds.setncattr(key, value)
+            if self.is_list_of_strings(value) and (self.format == 'NETCDF4'):
```

Comment:
We probably want to use this in all cases, even if a different format from NETCDF4 is selected. It's better to error than to improperly serialize data.

Reply:
OK, I tested with netCDF4 1.2.2 (last release before setncattr_string was introduced) and successfully catch the error with this latest commit. Is this along the lines of what you were thinking?